### PR TITLE
cql3: expr: allow bind markers in collection literals

### PR DESCRIPTION
**Pull request by havaker**
Allowing bind markers in collection literals is a change which causes minor differences in behavior between Scylla and Cassandra. Despite such an undesirable effect, I think allowing them is a good idea because it makes [refactoring work made by cvybhu](https://github.com/scylladb/scylla/pull/10409) easier - https://github.com/cvybhu/scylla/commit/469d03f8c2efda3ca9736f5a9e0169f02767bba4.

Also, making Scylla accept a superset of valid Cassandra cql expressions does not make us less compatible (maybe apart from test suit compatibility).

### CODE REVIEW

File: test/boost/cql_query_test.cc [hunk 1]
```diff
@@ -361,6 +361,51 @@ SEASTAR_TEST_CASE(test_list_of_tuples_with_bound_var) {
...
+                e.execute_prepared(stmt, {unset_value}).get(),
+                exceptions::invalid_request_exception
+            );
+
+            const auto int_value = cql3::raw_value::make_value(short_type->decompose(data_value(69)));
+            BOOST_REQUIRE_THROW(
+                e.execute_prepared(stmt, {int_value}).get(),
+                exceptions::invalid_request_exception
+            );
+        }
```
Comment (nyh) [inline test/boost/cql_query_test.cc:405]: What about testing a **successful** case?
If the goal of this test is only testing errors, maybe you should clarify this in its name and/or a comment.

**Comment by havaker** [comment]
v2:

- test a successful case

**Comment by psarna** [comment]
This commit would allow a nice simplification in #10409 - @havaker please ping me once this PR is moved out of draft and a description is provided.

**Comment by havaker** [comment]
> This commit would allow a nice simplification in #10409 - @havaker please ping me once this PR is moved out of draft and a description is provided.

@psarna ping

**Review by psarna — APPROVED** [review]

**Comment by psarna** [comment]
@nyh if your comments were addressed in a satisfactory way, feel free to proceed with enqueuing this one

File: test/boost/cql_query_test.cc [hunk 2]
```diff
@@ -357,7 +357,7 @@ SEASTAR_TEST_CASE(test_tuple_elements_validation) {
 SEASTAR_TEST_CASE(test_list_of_tuples_with_bound_var) {
     return do_with_cql_env_thread([](cql_test_env& e) {
         e.execute_cql("create table cf (pk int PRIMARY KEY, c1 list<frozen<tuple<int,int>>>);").get();
-        BOOST_REQUIRE_THROW(e.prepare("update cf SET c1 = c1 + [(?,9999)] where pk = 999;").get0(), exceptions::invalid_request_exception);
+        e.prepare("update cf SET c1 = c1 + [(?,9999)] where pk = 999;").get0();
```
Comment (nyh) [inline test/boost/cql_query_test.cc:360]: Again, it's weird to see a test that does an update without reading back and checking that the update actually worked.

**Review by nyh — APPROVED** [review]
Thanks. I'll commit. I have a couple of comments (actually both the same comment) on how the tests can be improved, but this can be done later.

**Pull request closed without merging**